core/lib/via_btc_client/src/client/mod.rs

```diff
@@ -67,11 +67,7 @@ impl BitcoinOps for BitcoinClient {
     }
 
     async fn check_tx_confirmation(&self, txid: &Txid) -> BitcoinClientResult<bool> {
-        let latest_block_hash = self.rpc.get_best_block_hash().await?;
-        let tx_info = self
-            .rpc
-            .get_raw_transaction_info(txid, Some(&latest_block_hash))
-            .await?;
+        let tx_info = self.rpc.get_raw_transaction_info(txid).await?;
 
         match tx_info.confirmations {
             Some(confirmations) => Ok(confirmations > 0),
```

Comment:
https://en.bitcoin.it/wiki/Irreversible_Transactions
https://developers.circle.com/circle-mint/docs/blockchain-confirmations#:~:text=To%20protect%20against%20reorg%20attacks,block%20may%20be%20invalidated%20increases.

this line assume if the tx get included in a block we consider it in confirmed 
but confirmation is something relative to level of security we want.
because of that suggest add new input to this method with name required confirmation replace it the 0 with that input (the community standard now is 3 but it's not static number every business should decide about this by themselves)

Reply:
```
    async fn estimate_fee(&self, conf_target: u16) -> BitcoinClientResult<u64> {
        let estimation = self
            .rpc
            .estimate_smart_fee(conf_target, Some(EstimateMode::Economical))
            .await?;
        match estimation.fee_rate {
            Some(fee_rate) => Ok(fee_rate.to_sat()),
            None => {
                let err = match estimation.errors {
                    Some(errors) => errors.join(", "),
                    None => "Unknown error during fee estimation".to_string(),
                };
                Err(BitcoinError::FeeEstimationFailed(err))
            }
        }
    }
}
```

what is the `conf_target` in input of this method?
this method just return fee_rate, let name it get_fee_rate or something like that instead of estimate_fee

Reply:
it stands for confirmation target also: https://developer.bitcoin.org/reference/rpc/estimatesmartfee.html